lib/accesslists/hierarchy_legacy.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package accesslists
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/accesslist"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// GetMembersFor returns a flattened list of Members for an Access List, including inherited Members.
+//
+// Returned Members are not validated for expiration or other requirements – use IsAccessListMember
+// to validate a Member's membership status.
+// DEPRECATED: use Hierarchy.GetMembersFor instead.
+func GetMembersFor(ctx context.Context, accessListName string, g AccessListAndMembersGetter) ([]*accesslist.AccessListMember, error) {
+	h, err := NewHierarchy(HierarchyConfig{
+		AccessListsService: g,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	members, err := h.GetMembersFor(ctx, accessListName)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return members, nil
+}
+
+// GetOwnersFor returns a flattened list of Owners for an Access List, including inherited Owners.
+//
+// Returned Owners are not validated for expiration or other requirements – use IsAccessListOwner
+// to validate an Owner's ownership status.
+// DEPRECATED: use Hierarchy.GetOwnersFor instead.
+func GetOwnersFor(ctx context.Context, accessList *accesslist.AccessList, g AccessListAndMembersGetter) ([]*accesslist.Owner, error) {
+	h, err := NewHierarchy(HierarchyConfig{
+		AccessListsService: g,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	owners, err := h.GetOwnersFor(ctx, accessList)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return owners, nil
+}
+
+// IsAccessListOwner checks if the given user is the Access List owner. It returns an error matched
+// by [IsUserLocked] if the user is locked.
```

Comment:
> It returns an error matched by [IsUserLocked] if the user is locked.
```suggestion
// IsAccessListOwner checks if the given user is the Access List owner. It may return an error matched
// by [IsUserLocked] if the user is locked.
```

Otherwise, I found it a bit misleading cause an error is returned for `NewHierarchy` too, which may not have the `IsUserLocked`.